hazelcast/test/src/HazelcastTests7.cpp

```diff
@@ -530,6 +530,35 @@ TEST_F(ClientListTest, testListener)
     ASSERT_TRUE(list->remove_item_listener(registrationId).get());
 }
 
+TEST_F(ClientListTest, testListenerOnRemoved)
+{
+    boost::latch latch1(1);
+
+    item_listener listener;
+
+    list->add("item-1").get();
+
+    listener.on_removed([&latch1](item_event&& item_event) {
+        auto type = item_event.get_event_type();
+        ASSERT_EQ(item_event_type::REMOVED, type);
+        ASSERT_EQ("MyList", item_event.get_name());
+        std::string host = item_event.get_member().get_address().get_host();
+        ASSERT_TRUE(host == "localhost" || host == "127.0.0.1");
+        ASSERT_EQ(5701, item_event.get_member().get_address().get_port());
+        ASSERT_EQ("item-1", item_event.get_item().get<std::string>().value());
+        latch1.count_down();
+    });
+
+    auto registrationId =
+      list->add_item_listener(std::move(listener), true).get();
+
+    list->remove("item-1").get();
+
+    ASSERT_OPEN_EVENTUALLY(latch1);
+
+    ASSERT_TRUE(list->remove_item_listener(registrationId).get());
```

Comment:
Is `remove_item_listener` required ? I think it can be removed.

```suggestion
```

Reply:
not a must, but while we are adding a listener, it will be nice to remove it before the test case end.